3-With-Intune MAM/IntuneClient/active-directory-xamarin-intune/PCAWrapper.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Xamarin.Forms;
+
+namespace activedirectoryxamarinintune
+{
+    /// <summary>
+    /// This is a wrapper for PCA. It is singleton and can be utilized by both application and the MAM callback
+    /// </summary>
+    public class PCAWrapper
+    {
+
+        /// <summary>
+        /// This is the singleton used by consumers
+        /// </summary>
+        static public PCAWrapper Instance { get; }
+
+        internal IPublicClientApplication PCA { get; }
+
+        // This event coordinates between different callbacks
+        public static ManualResetEvent MAMRegsiteredEvent { get; } = new ManualResetEvent(false);
+
+        /// <summary>
+        /// The authority for the MSAL PublicClientApplication. Sign in will use this URL.
+        /// </summary>
+        private const string _authority = "https://login.microsoftonline.com/organizations";
+
+        // ClientID of the application in (msidlab4.com)
+        private static string _clientID = "94996db4-9d57-422d-a707-84a1328a3cb8"; // TODO - Replace with your client Id. And also replace in the AndroidManifest.xml
+
+        // TenantID of the organization (msidlab4.com)
+        private static string _tenantID = "f645ad92-e38d-4d1a-b510-d1b09a74a8ca"; // TODO - Replace with your TenantID. And also replace in the AndroidManifest.xml
+
+        static string[] clientCapabilities = { "ProtApp" }; // It is must to have these capabilities
```

Comment:
`private readonly static[] string clientCapabilities `

Reply:
This cannot be changed as readonly static is not initialized before the static constructor ref: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/static-constructors

Reply:
Please use Lazy<T> for singleton or make entire class static.